File: app/src/main/java/com/wojdor/hearthstonecards/application/util/CardImageDownloader.java
package com.wojdor.hearthstonecards.application.util;

import android.content.Context;
import android.graphics.Bitmap;

import com.bumptech.glide.Glide;

import java.util.concurrent.ExecutionException;

import timber.log.Timber;

public class CardImageDownloader {

    private static final String CARD_IMAGE_API = "https://art.hearthstonejson.com/v1/render/latest/%s/256x/%s.png";

    private Context context;
    private FileStorage fileStorage;

    public CardImageDownloader(Context context) {
        this.context = context;
        fileStorage = new FileStorage(context);
    }

    public void getImage(String fileName, String locale) {
        try {
            String url = String.format(CARD_IMAGE_API, locale, fileName);
            Bitmap bitmap = Glide.with(context)
                    .asBitmap()
                    .load(url)
                    .submit()
                    .get();
            bitmap.setHasAlpha(true);
            fileStorage.write(fileName, bitmap);
        } catch (InterruptedException | ExecutionException error) {
            Timber.e(error);
        }
    }
}

File: app/src/main/java/com/wojdor/hearthstonecards/application/util/CardImageDownloader.kt
package com.wojdor.hearthstonecards.application.util

import com.squareup.picasso.Picasso
import timber.log.Timber
import java.util.concurrent.ExecutionException

class CardImageDownloader(private val fileStorage: FileStorage) {

    fun getImage(fileName: String, locale: String) {
        try {
            val url = String.format(CARD_IMAGE_API, locale, fileName)
            val bitmap = Picasso.get()
                    .load(url)
                    .get()
            fileStorage.write(fileName, bitmap)
        } catch (error: IllegalStateException) {
            Timber.e(error)
        } catch (error: ExecutionException) {
            Timber.e(error)
        }
    }

    companion object {
        private const val CARD_IMAGE_API = "https://art.hearthstonejson.com/v1/render/latest/%s/256x/%s.png"
    }
}
